include/dos_system.h

```diff
@@ -343,10 +343,11 @@ class DOS_Drive {
 	DOS_Drive();
 	virtual ~DOS_Drive() = default;
 
-	virtual bool FileOpen(DOS_File** file, const char* name, uint8_t flags) = 0;
-	virtual bool FileCreate(DOS_File** file, const char* name,
-	                        FatAttributeFlags attributes) = 0;
-	virtual bool FileUnlink(const char* _name)=0;
+	virtual std::unique_ptr<DOS_File> FileOpen(const char* name,
+	                                           uint8_t flags) = 0;
```

Comment:
`const uint8_t flags` ?

Reply:
It gets modified inside the functions.  Apparently I can lie in and state that it's const inside the header file and it still compiles with no errors/warnings interestingly enough.  Only the `const` in the actual function definition matters.

Reply:
Yup, that's how C/C++ works. Passing by value creates a copy, so it does not matter if function definition uses it as a const value or not. This also means that we can always make any parameter passed by value a const without modifying signature of the function (which is a good thing). One of those obscure C/C++ corners…

Reply:
@dreamer Yeah we had some discussion on that a while back.  I knew it worked the other way (const in the definition, non-const in the declaration).  I was a bit surprised by this though, considering it's also a virtual function.  Makes sense though.

I'm in favor of leaving these value types as-is then considering it doesn't even enforce that the argument is const in the implementation and seemingly makes no difference at all.

Reply:
Yeah, if you don't wanna refactor this whole mess further, that's fine @weirddan455 

But in general, to recap what we agreed on previously:

1. We use const both in function definitions and declarations for consistency. Having it in one place is a bit too clever, confusing, and not worth it, agreed.
2. We use const for local variables where applicable. The same reasoning that applies to making local vars const also applied to incoming function arguments, so we make them const whenever possible in new code.
3. When refactoring old code, yeah, fixing up function definitions and declarations it's up to the pain tolerance of the person doing the refactoring 😆